File: golang/client/queries/getClientsQueryHandler.go
package client

import (
	client "github.com/houstondapaz/cqrs-govscsharp/golang/client/domain"
	queries "github.com/houstondapaz/cqrs-govscsharp/golang/domain/queries"
)

type GetClientsQueryHandler struct {
	queries.QueryHandler
	storage client.ClientStorager
}

func NewGetClientsQueryHandler(storage client.ClientStorager) GetClientsQueryHandler {
	return GetClientsQueryHandler{
		storage: storage,
	}
}
func (qh GetClientsQueryHandler) QueryName() string {
	return "GetClientsQuery"
}

func (qh GetClientsQueryHandler) Handle(query queries.Querier) (interface{}, error) {
	return qh.storage.GetAll()
}

File: csharp/src/CQRSService.Client/Queries/GetClientsQueryHandler.cs
using CQRSService.Client.Domain.Queries;
using CQRSService.Client.Storage;
using CQRSService.Domain.Queries;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace CQRSService.Client.Queries
{
    public class GetClientsQueryHandler : IQueryHandler<GetClientsQuery, IEnumerable<Client>>
    {
        private readonly IStorage<Client> _clientStorage;

        public GetClientsQueryHandler(IStorage<Client> clientStorage) => _clientStorage = clientStorage;
        public Task<IEnumerable<Client>> Handle(GetClientsQuery request, CancellationToken cancellationToken)
            => _clientStorage.GetAll();
    }
}
